### DISCUSSION TITLE
Root user used to run Milvus in containers

### DISCUSSION THREAD

**Discussion by sskserk** [milvus-io/milvus#34163] [Q&A and General discussion]
Hi Milvus developers,

Thank you for the great solution. In fact, your product is almost the only alternative nowadays.

We are looking to go into production with a Milvus cluster deployed with Helm. It just works... However, it always happens that from a functional perspective everything is fine, but when it comes to security measures, a couple of possible issues appear. This time, our security scanners were triggered by the root user which the Milvus processes run under.

Does it seem feasible to change the user to a less privileged one?

Thank you!

**Comment by yhmo** [comment]
Do you mean to change the privileges of the root user?  I don't think it is possible, it is a build-in account.

**Reply by sskserk** [reply]
No, just change the user which is used by Milvus processes running in containers https://medium.com/@mccode/processes-in-containers-should-not-run-as-root-2feae3f0df3b

**Reply by LoveEachDay** [reply]
@sskserk Run as non-root user would be a nice feature for milvus-helm chart to support. We would like to add this feature. 

For now, in the milvus-helm chart would generate two config files: user.yaml and default.yaml. And before the process startup, it would merge those two config files with milvus.yaml in the image, and then write the merged configs back into milvus.yaml which would require root privilege.

**Reply by sskserk** [reply]
@LoveEachDay,
any plans to implement it in the nearest future? ;-)

I believe many customers will benefit from this feature.
